docs/book/src/tasks/upgrading-cluster-api-versions.md

```diff
@@ -0,0 +1,123 @@
+This should be the case for all infrastructure providers using conversion webhooks to allow upgrading from `v1alpha2` to
+`v1alpha3`.
+
+This can verified by running `kubectl get crd <crd name>.infrastructure.cluster.x-k8s.io -o yaml` for all the
+infrastructure provider CRDs.
+
+### Upgrade Cluster API components using clusterctl
+
+Run [clusterctl init] with the relevant infrastructure flag. For the [AWS provider] you would run:
+
+```bash
+clusterctl init --infrastructure aws
+```
+
+You should now be able to manage your resources using the `v1alpha3` version of the Cluster API components.
+
+### Adopting existing machines into KubeadmControlPlane management
+
+<aside class="note warning">
+
+<h1> Important </h1>
+
+You must be running at least Cluster API 0.3.7 for adoption to be successful.
+
+</aside>
+
+If your cluster has existing machines labeled with `cluster.x-k8s.io/control-plane`, you may opt in to management of those machines by
+creating a new KubeadmControlPlane object and updating the associated Cluster object's `controlPlaneRef` like so:
+
+```
+---
+apiVersion: "cluster.x-k8s.io/v1alpha3"
+kind: Cluster
+...
+spec:
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+    kind: KubeadmControlPlane
+    name: controlplane
+    namespace: default
+...
+```
+
+Caveats:
+
+* The KCP controller will refuse to adopt any control plane Machines not bootstrapped with the kubeadm bootstrapper.
+* The KCP controller may immediately begin upgrading Machines post-adoption if they're out of date.
+* The KCP controller attempts to behave intelligently when adopting existing Machines, but because the bootstrapping process sets various fields in the KubeadmConfig of a machine it's not always obvious the original user-supplied `KubeadmConfig` would have been for that machine. The controller attempts to guess this intent to not replace Machines unnecessarily, so if it guesses wrongly, the consequence is that the KCP controller will effect an "upgrade" to its current config. For full details, see SemanticMerge in the kubeadm bootstrapper's api/equality package.
+* If the cluster's PKI materials were generated by an initial KubeadmConfig reconcile, they'll be owned by the KubeadmConfig bound to that machine. The adoption process re-parents these resources to the KCP so they're not lost during an upgrade, but deleting the KCP post-adoption will destroy those materials.
+* The `ClusterConfiguration`  is not currently reconciled with their ConfigMaps the workload cluster, and `kubeadm` considers the ConfigMap authoritative. These fields on the KCP will be effectively ignored, and most notably include:
+    * `kubeadmConfigSpec.clusterConfiguration.apiServer.extraArgs`
+    * `kubeadmConfigSpec.clusterConfiguration.controllerManager.extraArgs`
+    * `kubeadmConfigSpec.clusterConfiguration.scheduler.extraArgs`
+    * Anything underneath `kubeadmConfigSpec.clusterConfiguration.etcd`
```

Comment:
We do allow changes under clusterConfiguration to .etcd.local.{imageRepository,imageTag}, .dns.{imageRepository,imageTag}, and .imageRepository. Worth calling out? Also probably worth linking to the issue about full mutability - #2083.

Reply:
Thanks for that. I didn't notice that was out of date when I shuffled the copy around.